apps/api/test/08-report-8.e2e-spec.ts

```diff
@@ -0,0 +1,131 @@
+  let app: INestApplication;
+  let applicantId: string;
+  let HA: IENHaPcn[] = [];
+
+  const lastYear = dayjs().year() - 1;
+  const fiscalDate = dayjs(`${lastYear}-06-06`).format('YYYY-MM-DD');
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideGuard(AuthGuard)
+      .useValue({ canActivate })
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    setApp(app);
+    HA = await getHAs();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  const getReport8 = async () => {
+    const { body } = await request(app.getHttpServer()).get(URLS.REPORT8);
+    return body;
+  };
+
+  it('Add status to fiscal year', async () => {
+    const before = await getReport8();
+
+    const applicant = getApplicant();
+    applicant.registration_date = '2022-06-01';
+    const { id } = await addApplicant(applicant);
+
+    await hire(id, 'FNHA', dayjs().format('YYYY-MM-DD'));
+
+    await addMilestone(id, '', {
+      status: await getStatusId(STATUS.RECEIVED_WORK_PERMIT),
+      start_date: fiscalDate,
+    });
+
+    const after = await getReport8();
+
+    // HA count
+    expect(Number(after[0]['current_fiscal'])).toBe(Number(before[0]['current_fiscal'] + 1));
+    // total count
+    expect(Number(after[HA.length]['current_fiscal'])).toBe(
+      Number(before[HA.length]['current_fiscal'] + 1),
+    );
+    expect(Number(after[HA.length]['total'])).toBe(Number(before[HA.length]['total'] + 1));
+  });
+
+  it('Add status to current period', async () => {
+    const before = await getReport8();
+
+    const applicant = getApplicant();
+    applicant.registration_date = '2022-06-01';
+    const { id } = await addApplicant(applicant);
+    applicantId = id;
+
+    await hire(id, 'FHA', dayjs().format('YYYY-MM-DD'));
+
+    await addMilestone(id, '', {
+      status: await getStatusId(STATUS.RECEIVED_WORK_PERMIT),
+      start_date: dayjs().format('YYYY-MM-DD'),
+    });
+    const after = await getReport8();
+
+    // HA count
+    expect(Number(after[1]['current_fiscal'])).toBe(Number(before[1]['current_fiscal']) + 1);
+    // total count
+    expect(Number(after[HA.length]['current_fiscal'])).toBe(
+      Number(before[HA.length]['current_fiscal']) + 1,
+    );
+    expect(Number(after[HA.length]['total'])).toBe(Number(before[HA.length]['total']) + 1);
+  });
+
+  it('Add duplicate status to different job for same applicant', async () => {
+    const before = await getReport8();
+    await hire(applicantId, 'IHA', dayjs().format('YYYY-MM-DD'));
+
+    await addMilestone(applicantId, '', {
+      status: await getStatusId(STATUS.RECEIVED_WORK_PERMIT),
+      start_date: fiscalDate,
+    });
+    const after = await getReport8();
+
+    // same applicant, should count only highest/ most recent work permit milestone
+    expect(Number(after[1]['total'])).toBe(Number(before[1]['total']) - 1);
```

Comment:
Why does the second HA's total decrease?

Reply:
the docs say the report is supposed to take the most recent milestone, assuming to prevent duplicates in the data.  Just an example, if the Received Work Permit milestone is added twice, it won't count it as 2.  Not sure how frequent that would happen, or if it will.  Can remove the test if we deem it unnecessary

Reply:
I'd have to double check, but don't we take the 'latest' milestone in a few other reports as well

Reply:
Let's say that an applicant got a work permit in 2020.1 and reporting period is 2021.1 ~ 2022.1.
Then, the work permit got reissued in 2023.1. If we count only the latest one, then the applicant is considered as not working during the period.